pkg/commands/pull_request.go

```diff
@@ -2,72 +2,81 @@ package commands
-			},
-			pullRequestURLIntoTargetBranch: func(owner string, repository string, from string, to string) string {
-				return fmt.Sprintf("https://%s/%s/%s/merge_requests/new?merge_request[source_branch]=%s&merge_request[target_branch]=%s", siteDomain, owner, repository, from, to)
-			},
-		}
+func NewGithubService(repositoryDomain string, siteDomain string) *Service {
+	return &Service{
+		Name: repositoryDomain,
+		pullRequestURLIntoDefaultBranch: func(owner string, repository string, from string) string {
+			return fmt.Sprintf("https://%s/%s/%s/compare/%s?expand=1", siteDomain, owner, repository, from)
+		},
+		pullRequestURLIntoTargetBranch: func(owner string, repository string, from string, to string) string {
+			return fmt.Sprintf("https://%s/%s/%s/compare/%s...%s?expand=1", siteDomain, owner, repository, to, from)
+		},
+		URLRegexStrings: defaultUrlRegexStrings,
 	}
+}
 
-	return service
+func NewBitBucketService(repositoryDomain string, siteDomain string) *Service {
+	return &Service{
+		Name: repositoryDomain,
+		pullRequestURLIntoDefaultBranch: func(owner string, repository string, from string) string {
+			return fmt.Sprintf("https://%s/%s/%s/pull-requests/new?source=%s&t=1", siteDomain, owner, repository, from)
+		},
+		pullRequestURLIntoTargetBranch: func(owner string, repository string, from string, to string) string {
+			return fmt.Sprintf("https://%s/%s/%s/pull-requests/new?source=%s&dest=%s&t=1", siteDomain, owner, repository, from, to)
+		},
+		URLRegexStrings: defaultUrlRegexStrings,
+	}
 }
 
-func (s *Service) PullRequestURL(owner string, repository string, from string, to string) string {
+func NewGitLabService(repositoryDomain string, siteDomain string) *Service {
+	return &Service{
+		Name: repositoryDomain,
+		pullRequestURLIntoDefaultBranch: func(owner string, repository string, from string) string {
+			return fmt.Sprintf("https://%s/%s/%s/merge_requests/new?merge_request[source_branch]=%s", siteDomain, owner, repository, from)
+		},
+		pullRequestURLIntoTargetBranch: func(owner string, repository string, from string, to string) string {
+			return fmt.Sprintf("https://%s/%s/%s/merge_requests/new?merge_request[source_branch]=%s&merge_request[target_branch]=%s", siteDomain, owner, repository, from, to)
+		},
+		URLRegexStrings: defaultUrlRegexStrings,
+	}
+}
+
+func (s *Service) PullRequestURL(repoURL string, from string, to string) string {
+	repoInfo := s.getRepoInfoFromURL(repoURL)
+
 	if to == "" {
-		return s.pullRequestURLIntoDefaultBranch(owner, repository, from)
+		return s.pullRequestURLIntoDefaultBranch(repoInfo.Owner, repoInfo.Repository, from)
 	} else {
-		return s.pullRequestURLIntoTargetBranch(owner, repository, from, to)
+		return s.pullRequestURLIntoTargetBranch(repoInfo.Owner, repoInfo.Repository, from, to)
 	}
 }
 
 // PullRequest opens a link in browser to create new pull request
 // with selected branch
 type PullRequest struct {
-	GitServices []*Service
```

Comment:
Given how cheap it is to obtain these I'm not storing them on the pull request struct